docs/src/main/asciidoc/tls-registry-reference.adoc

```diff
@@ -149,6 +150,20 @@ quarkus.http.tls-configuration-name=MY_TLS_CONFIGURATION
 quarkus.grpc.clients.hello.tls-configuration-name=MY_TLS_CONFIGURATION
 ----
 
+.Example configuration for a SmallRye GraphQL client:
+[source,properties]
+----
+quarkus.smallrye-graphql-client.my-client.tls-configuration-name=MY_TLS_CONFIGURATION
+----
+
+[NOTE]
+====
+When using the Typesafe GraphQL client with a certificate reloading mechanism (see <<reloading-certificates>>),
+it is essential to override the bean's scope to `RequestScoped` (or another similar scope). By default, the
+Typesafe client is an application-scoped bean.  This guarantees that each new instance of the bean created
+after a certificate reload will be configured with the latest certificate.
```

Comment:
Not sure if it's just me misunderstanding but this looks confusing to me.

AFAICS from the first sentence, `ApplicationScoped` beans won't work well.

But the second sentence says some beans are application scoped and then it says `This guarantees...`. So you might think that the fact that they are application scoped guarantees it to work. Which is not the case unless I'm mistaken?

Also is it a general limitation of the TLS registry because it seems quite problematic to require all the beans to be request scoped?